packages/flutter_tools/lib/src/vscode/vscode_validator.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../base/version.dart';
+import '../doctor.dart';
+import 'vscode.dart';
+
+class VsCodeValidator extends DoctorValidator {
+  final VsCode _vsCode;
+
+  VsCodeValidator(this._vsCode) : super('VS Code');
+
+  static Iterable<DoctorValidator> get installedValidators {
+    return VsCode
+        .allInstalled()
+        .map((VsCode vsCode) => new VsCodeValidator(vsCode));
+  }
+
+  @override
+  Future<ValidationResult> validate() async {
+    final List<ValidationMessage> messages = <ValidationMessage>[];
+    ValidationType type = ValidationType.missing;
+    final String vsCodeVersionText = _vsCode.version == Version.unknown
+        ? null
+        : 'version ${_vsCode.version}';
+    messages.add(new ValidationMessage('VS Code at ${_vsCode.directory}'));
```

Comment:
We may not need to print the location of the VSCode install - I think it's pretty uniform (IntelliJ / Android Studio does tend to get installed in several places, so probably still useful for that tool). I guess the criteria for whether we display the vscode location is whether we think it'd be useful for users debugging issues, or use debugging users reported issues.

Reply:
I'm OK with printing it here; helps to be consistent with the other doctors as well as to give the user some confidence.

Reply:
I think there's less value for VS Code because we only look in default install locations; we don't really scan around (or support overriding the path). However I do think it may come in handy for debugging (esp on Windows where there are more options - there's 64bit + 32bit and also the paths aren't fixed since they come from progfiles env vars) and since doctor output is quite terse by default now, I don't think it's adding noise.

(ofc, I don't feel strongly either way, so happy to remove if requested)

Reply:
keeping the directory sgtm; it makes sense for consistency w/ the other validators